capstone/src/main/java/com/google/step/servlets/RestaurantsServlet.java

```diff
@@ -14,55 +14,66 @@
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import org.json.JSONArray;
-import org.json.JSONObject;
 
-/** Servlet responsible for getting restaurants from Datastore. */
+/** Servlet responsible for getting restaurants from the search index */
 @WebServlet("/restaurants")
 public class RestaurantsServlet extends HttpServlet {
-  private RestaurantClient restaurantClient = new RestaurantClient();
+  private final RestaurantHeaderSearchClient searchClient;
+
+  public RestaurantsServlet() {
+    this(new ElasticsearchClient("localhost", (short)9200));
+  }
+
+  RestaurantsServlet(RestaurantHeaderSearchClient searchClient) {
+    this.searchClient = searchClient;
+  }
 
   /**
-   * Returns a list of RestaurantHeaders, with a snapshot of restaurant details.
-   * @param response A list of restaurant details, in the following json format:
-    {
-      "restaurantHeaders" : [
-        {
-          "name": <String>,
-          "cuisine": <List<String>>,
-          "message": <String, details status of restaurant based on score>,
-        },
-        ...
-      ]
-    }
-   * @param request Specifies relevant search params. //TODO: update exact search param format
-   * @throws //TODO: add any error codes thrown once search is implemented.
+   * Request headers from the search server for restaurants without any particular query
+   * @param response  a servlet response with a JSON object holding restaurant details in the
+   *                 response body. The restaurant details can be found in the "restaurants" field
+   * @param request simple get request. Takes no parameters
+   * @throws IOException thrown if writing the response fails
    */
-  // TODO: Change this to be connected to search functionality.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Restaurant> restaurants = restaurantClient.getRestaurantsNoFilter();
+    List<RestaurantHeader> restaurantHeaders = searchClient.searchRestaurants("");
+    JSONObject responseJson = new JSONObject()
+        .put("restaurants", new JSONArray(restaurantHeaders));
+
+    response.setContentType(Json.MEDIA_TYPE);
+    response.getWriter().println(responseJson);
+  }
 
-    // Format restaurant List to JSON for return
-    JSONArray restaurantsListJson = new JSONArray(restaurants);
+  /**
+   * Query the search server for a specific subset of matching restaurants
+   * @param request servlet request with "query" parameter set to search query
+   * @param response  servlet response with a JSON body. The "restaurants" field of the body
+   *                  contains an array of restaurant details JSON objects
+   * @throws IOException thrown if writing the response fails
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Why are we adding this method, instead of making doGet support a parameter?

Reply:
I changed the servlet to only use `doGet()`.